controller/lib/debug/vars.h

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Defines an interface for creating debug variables.
+//
+// All debug variables must have a 32-bit value: int32_t, uint32_t, or float.
+// They are declared using DebugVar that takes a name, help string,
+// printf format string, and getter and setter lambdas:
+//
+//   DebugVar var("speed", "The vehicle speed", "%.3f",
+//     [&] { return ...; },
+//     [&](float value) { ... });
+//
+// All debug variables in the program are accessible via a static registry:
+//
+//   DebugVarRegistry::GetVar(i) is valid for all i in [0, N)
+//   where N = DebugVarRegistry::GetNumVars().
+//
+#ifndef VARS_H
+#define VARS_H
+
+#include "debug.h"
+#include <stdint.h>
+#include <type_traits>
+
+// Defines the type of variable
+enum class VarType {
+  INT32 = 1,
+  UINT32 = 2,
+  FLOAT = 3,
+};
+
+#define ARRAY_CT(a) (sizeof(a) / sizeof((a)[0]))
```

Comment:
This Classic C Macro is evil, because you can do `int *x; ARRAY_CT(x)` and it will give you...something.

There's a Classic C++ Template Incantation for it, and TIL it's in the STL now as `std::size`, https://en.cppreference.com/w/cpp/iterator/size

But also, do you need this?  Or can you just make 100 a constant in DebugVarRegistry?